File: ch5-ex1/src/main/java/study/spring/security/ch5ex1/config/ProjectConfig.java
package study.spring.security.ch5ex1.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authentication.AuthenticationProvider;
import org.springframework.security.config.Customizer;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
public class ProjectConfig {

  private final AuthenticationProvider authenticationProvider;

  public ProjectConfig(AuthenticationProvider authenticationProvider) {
    this.authenticationProvider = authenticationProvider;
  }

  @Bean
  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
    http
        .httpBasic(Customizer.withDefaults())
        .authorizeHttpRequests(customizer -> customizer.anyRequest().authenticated())
        .authenticationProvider(authenticationProvider);

    return http.build();
  }

}

File: ch9-ex1/src/main/kotlin/study/spring/security/ch9ex1/config/ProjectConfig.kt
package study.spring.security.ch9ex1.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.SecurityFilterChain
import org.springframework.security.web.authentication.www.BasicAuthenticationFilter

@Configuration
class ProjectConfig {

    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        http
            .addFilterBefore(
                RequestValidationFilter(),
                BasicAuthenticationFilter::class.java)
            .addFilterAfter(
                AuthenticationLoggingFilter(),
                BasicAuthenticationFilter::class.java)
            .authorizeHttpRequests {
                it.anyRequest().permitAll()
            }

        return http.build()
    }
}